client/jetpack-connect/signup.js

```diff
@@ -98,6 +98,7 @@ export class JetpackSignup extends Component {
 
 	handleSubmitSignup = ( _, userData, analyticsData, afterSubmit = noop ) => {
 		debug( 'submitting new account', userData );
+		userData.extra = { jpc: true };
 		this.setState( { isCreatingAccount: true }, () =>
 			this.props
 				.createAccount( userData )
```

Comment:
Could we add the `extra: { jpc: true }` prop here instead, to avoid the mutation? Like this:

```
.createAccount( { ...userData, extra: { jpc: true } } )
```

Reply:
Yes, thanks for the review and suggestion. Updated in the latest commit.